src/core/components/Experiment/index.js

```diff
@@ -0,0 +1,158 @@
+/* @flow */
+import * as React from 'react';
+import invariant from 'invariant';
+import cookie from 'react-cookie';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import withUIState from 'core/withUIState';
+import tracking from 'core/tracking';
+import type { AppState } from 'amo/store';
+
+type Props = {|
+  AName: string,
+  BName: string,
+  nameId: string,
+|};
+
+type UIStateType = {|
+  abTest: string | void,
```

Comment:
Why do you need a ui state btw? And why a "ui state" and not a regular redux state?

Reply:
well I *think* this would have been a case where I used regular state (b/c of situation where class wasn't assigned on client on initial visit) here but thought we were trying to use UIState instead..? I'll revisit this..